File: _scratch/run/decode_string/sol.py
class Solution:
    def decodeString(self, s):
        st = []
        cur = ""
        num = 0
        for c in s:
            if c.isdigit():
                num = num * 10 + int(c)
            elif c == "[":
                st.append((cur, num))
                cur = ""
                num = 0
            elif c == "]":
                prev, k = st.pop()
                cur = prev + cur * k
            else:
                cur += c
        return cur
import sys
_l=sys.stdin.read().split('\n')
_T=int(_l[0]);_o=[]
for _i in range(_T):
    _s=_l[1+_i] if 1+_i<len(_l) else ""
    _o.append(str(Solution().decodeString(_s)))
print('\n'.join(_o))

File: _scratch/run/decode_string/sol.js
/**
 * @param {string} s
 * @return {string}
 */
var decodeString = function(s) {
    const st = [];
    let cur = "", num = 0;
    for (const c of s) {
        if (c >= '0' && c <= '9') num = num * 10 + (+c);
        else if (c === '[') { st.push([cur, num]); cur = ""; num = 0; }
        else if (c === ']') { const [prev, k] = st.pop(); cur = prev + cur.repeat(k); }
        else cur += c;
    }
    return cur;
};
const _l=require('fs').readFileSync(0,'utf8').split('\n');const _T=+_l[0];const _o=[];
for(let _i=0;_i<_T;_i++){const _s=(_l[1+_i]||'').replace(/\r$/,'');_o.push(String(decodeString(_s)));}
console.log(_o.join('\n'));